leshan-core/src/main/java/org/eclipse/leshan/core/node/codec/json/LwM2mNodeJsonDecoder.java

```diff
@@ -94,72 +94,81 @@ private static LwM2mNode parseJSON(JsonRootObject jsonObject, LwM2mPath path, Lw
             Integer resourceId = Integer.valueOf(resourcePath[0]);
 
             if (!multiResourceMap.isEmpty() && multiResourceMap.get(resourceId) != null) {
```

Comment:
This "if" block could be part of the "if(resourcePath.length > 1)" block below. For readability.